File: irrgarten-java/src/irrgarten/Shield.java
package irrgarten;

/**
 * Escudo usado por el jugador durante el combate
 *
 * @author vik
 */
public class Shield {

    private float protection;
    private int uses;

    public Shield(float protection, int uses) {
        this.protection = protection;
        this.uses = uses;
    }

    /**
     * Intensidad de la defensa del jugador
     *
     * @return
     */
    public float protect() {
        float prot = 0;
        if (uses > 0) {
            prot = protection;
            uses--;
        }
        return prot;
    }

    @Override
    public String toString() {
        return String.format("S[%f.2, %d]", protection, uses);
    }
    
    public boolean discard() {
        return Dice.discardElement(uses);
    }
}

File: irrgarten-ruby/shield.rb
# encoding: utf-8

require_relative 'dice'

module Irrgarten

  class Shield
    def initialize(protection, uses)
      @protection = protection
      @uses = uses
    end

    def protect
      out = 0.0
      if @uses > 0
        out = @protection
        @uses -= 1
      end
      out
    end

    def discard
      Dice.discard_element(@uses)
    end

    def to_s
      "S[#{@protection}, #{@uses}]"
    end

  end

end

